lib/jekyll-redirect-from/redirector.rb

```diff
@@ -4,66 +4,70 @@ class Redirector < Jekyll::Generator
 
     def generate(site)
       original_pages = site.pages.dup
-      generate_alt_urls(site, site.posts) if Jekyll::VERSION < '3.0.0'
+      generate_alt_urls(site, site.posts) if Jekyll::VERSION < "3.0.0"
       generate_alt_urls(site, original_pages)
       generate_alt_urls(site, site.docs_to_write)
     end
 
     def generate_alt_urls(site, list)
       list.each do |item|
-        if has_alt_urls?(item)
+        if alt_urls?(item)
           alt_urls(item).each do |alt_url|
             redirect_page = RedirectPage.new(site, site.source, "", "redirect.html")
-            redirect_page.data['permalink'] = alt_url
+            redirect_page.data["permalink"] = alt_url
             redirect_page.generate_redirect_content(redirect_url(site, item))
             site.pages << redirect_page
           end
         end
-        if has_redirect_to_url?(item)
-          redirect_to_url(item).flatten.each do |alt_url|
-            item.data['sitemap'] = false
-            item.data['layout'] = nil
+        next unless redirect_to_url?(item)
+        redirect_to_url(item).flatten.each do |alt_url|
+          item.data["sitemap"] = false
+          item.data["layout"] = nil
 
-            item.url << "index.html" if item.url.end_with?("/")
-            redirect_page = RedirectPage.new(site, site.source, File.dirname(item.url), File.basename(item.url))
+          item.url << "index.html" if item.url.end_with?("/")
+          redirect_page = RedirectPage.new(site, site.source,
+                          File.dirname(item.url), File.basename(item.url))
 
-            redirect_page.data['permalink'] = item.url
-            redirect_page.generate_redirect_content(alt_url)
-            if item.is_a?(Jekyll::Document)
-              item.content = item.output = redirect_page.content
-            else
-              site.pages << redirect_page
-            end
+          redirect_page.data["permalink"] = item.url
+          redirect_page.generate_redirect_content(alt_url)
+          if item.is_a?(Jekyll::Document)
+            item.content = item.output = redirect_page.content
+          else
+            site.pages << redirect_page
           end
         end
       end
     end
 
-    def is_dynamic_document?(page_or_post)
+    def dynamic_document?(page_or_post)
       page_or_post.is_a?(Jekyll::Page) ||
         page_or_post.is_a?(Jekyll::Document) ||
-        (Jekyll::VERSION < '3.0.0' &&
+        (Jekyll::VERSION < "3.0.0" &&
           page_or_post.is_a?(Jekyll::Post))
+
+      alias :is_dynamic_document? :dynamic_document?
```

Comment:
I believe you'll want these to be outside the method, immediately after the method's `end` block.